IdentityCore/src/cache/accessor/MSIDAccountCredentialCache.m

```diff
@@ -313,6 +313,51 @@ - (BOOL)removeCredentialsWithQuery:(nonnull MSIDDefaultCredentialCacheQuery *)ca
                                 error:error];
 }
 
+// Remove credentials
+- (BOOL)removeExpiredAccessTokensCredentialsWithQuery:(nonnull MSIDDefaultCredentialCacheQuery *)cacheQuery
+                          context:(nullable id<MSIDRequestContext>)context
+                            error:(NSError * _Nullable * _Nullable)error
+{
+    assert(cacheQuery);
+    cacheQuery.targetMatchingOptions = MSIDAny;
+    cacheQuery.matchAnyCredentialType = NO;
+    NSDate *currentDataAndTime = [NSDate date];
+    NSMutableArray *resultsToDelete = [NSMutableArray array];
+    NSString *className = NSStringFromClass(self.class);
+
+    MSID_LOG_WITH_CTX_PII(MSIDLogLevelInfo, context, @"(Default cache) Removing expired credentials with type %@, environment %@, realm %@, clientID %@, unique user ID %@", [MSIDCredentialTypeHelpers credentialTypeAsString:cacheQuery.credentialType], cacheQuery.environment, cacheQuery.realm, cacheQuery.clientId, MSID_PII_LOG_TRACKABLE(cacheQuery.homeAccountId));
+
+    if (cacheQuery.exactMatch)
```

Comment:
Why do we return NO when query is exactMatch ?

Reply:
@dorinung can you provide some context? My understanding is that when the query requires exact match, we don't perform removal during look up.

Reply:
@kaisong1990  That is correct. We only want this code to run for non exact matches.